lib/net.js

```diff
@@ -359,7 +364,7 @@ Socket.prototype._final = function(cb) {
   debug('_final: not ended, call shutdown()');
 
   // otherwise, just shutdown, or destroy() if not possible
-  if (!this._handle || !this._handle.shutdown) {
+  if (!this[kShouldShutdown] || !this._handle || !this._handle.shutdown) {
```

Comment:
I think the issue might be that this runs `destroy()` when it previously didn’t.

I can try it too (later), but maybe undoing this line change and adding `if (!this[kShouldShutdown]) return cb();` below works?

Reply:
It doesn't because the `writable: false` parameter that is passed in to the `net.Socket` constructor does control other behaviors internally, and it does not mean "this is half-open".
If you want to give it a shot, feel free to push here. Otherwise we revert to the `_final`  solution.